docs/references/javascript/user/password-management.mdx

```diff
@@ -1,39 +1,282 @@
+These methods on the [`User`](/docs/references/javascript/user/user) object help you manage a user's password.
+
+The following examples assume that you have followed the [quickstart](/docs/quickstarts/javascript) in order to add Clerk to your JavaScript application.
 
 ## `updatePassword()`
 
+Updates the user's password. Note that passwords must be at least 8 characters long.
+
 ```typescript
 function updatePassword: (params: UpdateUserPasswordParams) => Promise<User>;
 ```
 
-Updates the user's password.
-
 ### `UpdateUserPasswordParams`
 
-
 | Name | Type | Description |
 | --- | --- | --- |
 | `newPassword` | `string` | The user's new password. |
-| `currentPassword?` | `string` | The user's current password. |
+| `currentPassword` | `string` | The user's current password. |
 | `signOutOfOtherSessions?` | `boolean \| undefined` | If set to `true`, all sessions will be signed out. |
 
+### `updatePassword()` example
+
+<InjectKeys>
+
+<Tabs type="npm-script" items={['NPM module', '<script>']}>
+  <Tab>
+    <CodeBlockTabs options={["main.js", "index.html"]}>
+    ```js filename="main.js"
+    import Clerk from '@clerk/clerk-js';
+
+    // Initialize Clerk with your Clerk publishable key
+    const clerk = new Clerk('{{pub_key}}');
+    await clerk.load();
+
+    if (clerk.user) {
+      document
+        .getElementById("update-password-button")
+        .addEventListener("click", async function () {
+          const currentPassword = document.getElementById("current-password").value;
+          const newPassword = document.getElementById("new-password").value;
+          try {
+            await clerk.user.updatePassword({
+              currentPassword,
+              newPassword,
+            });
+          } catch (err) {
+            console.error(err);
+          }
+        });
+    } else {
+      document.getElementById("app").innerHTML = `
+        <div id="sign-in"></div>
+      `;
+
+      const signInDiv = document.getElementById("sign-in");
+
+      clerk.mountSignIn(signInDiv);
+    }
+    ```
+
+    ```html filename="index.html"
+    <!doctype html>
+    <html lang="en">
+      <head>
+        <meta charset="UTF-8" />
+        <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>Clerk + JavaScript App</title>
+      </head>
+      <body>
+        <div id="app"></div>
+
+        <h1>Update password</h1>
+        <input
+          type="password"
+          id="current-password"
+          placeholder="Current password"
+        />
+        <input
+          type="password"
+          id="new-password"
+          placeholder="New password"
+        />
+        <button id="update-password-button">Update password</button>
+
+        <script type="module" src="/main.js"></script>
+      </body>
+    </html>
+    ```
+    </CodeBlockTabs>
+  </Tab>
+
+  <Tab>
+  ```html filename="index.html"
```

Comment:
I know we told them to do the JS quickstart first at the top of the page, but this example can't actually work without the `<script>` from the quickstart, correct?

```
  <script
    async
    crossorigin="anonymous"
    data-clerk-publishable-key=your_publishable_key
    src="https://[your-domain].clerk.accounts.dev/npm/@clerk/clerk-js@latest/dist/clerk.browser.js"
    type="text/javascript"
  ></script>
  ```

If that's the case, we should add it to the example, so users can actually copy-paste it and have it work (though they must provide their domain and pub key of course)

Reply:
yeahhh I thought about this a lot as I wrote the examples - I'll have to update other JS examples so stay tuned for a fix PR !